instructor/compiler.py

```diff
@@ -0,0 +1,469 @@
+                'name': analysis,
+                'conditions': {
+
+                },
+                'identifiers': {}
+            }
+        ret[analysis]['conditions'][condition] = {
+            'name': condition,
+            'short_name': condition
+        }
+        if not ret[analysis]['conditions'][condition].has_key('identifiers'):
+            ret[analysis]['conditions'][condition]['identifiers'] = {}
+        identifiers = {}
+        for p in sp.microscopy_images.all():
+            ret[analysis]['identifiers']["SP_ID_{}".format(p.strain_protocol_id)] = 1
+            ret[analysis]['conditions'][condition]['identifiers']["SP_ID_{}".format(p.strain_protocol_id)] = 1
+    return ret
+
+
+def lysate_kinds(a):
+    ret = {}
+    if a.western_blot.has_whole_cell_lysate:
+        ret['whole'] = {'name': 'Whole Cell'}
+    if a.western_blot.has_cytoplasmic_fractination:
+        ret['cyto'] = {'name': 'Cytoplasm'}
+    if a.western_blot.has_nuclear_fractination:
+        ret['nuclear'] = {'name': 'Nuclear'}
+    return ret
+
+
+def secondary_anti_body(assignment):
+    western_blot = assignment.western_blot
+    secondary = {}
+    for a in western_blot.antibodies.all():
+        secondary[a.secondary] = {
+            'name': a.secondary
+        }
+    return secondary
+
+
+def primary_anti_body(assignment):
+    ret = {}
+    western_blot = assignment.western_blot
+    primary = {}
+    for a in western_blot.antibodies.all():
+        if primary.has_key(a.primary):
+            primary[a.primary].append(a.secondary)
+        else:
+            primary[a.primary] = [a.secondary]
+    order = {}
+    for a in western_blot.antibodies.all():
+        pk = 'AB_{}'.format(a.id)
+        order[pk] = 1
+        marks = []
+        for band in a.bands.all():
+            if western_blot.has_whole_cell_lysate:
+                marks.append({
+                    'weight': band.wcl_weight,
+                    'intensity': 0
+                })
+            if western_blot.has_cytoplasmic_fractination:
+                marks.append({
+                    'weight': band.cyto_weight,
+                    'intensity': 0
+                })
+            if western_blot.has_nuclear_fractination:
+                marks.append({
+                    'weight': band.nuc_weight,
+                    'intensity': 0
+                })
+        ret[pk] = {
+            'name': a.primary,
+            'secondary': primary[a.primary],
+            'marks': marks,
+            'gel_name': a.primary
+        }
+    ret['order'] = order.keys()
+    return ret
+
+
+def drugs(a):
+    ret = {}
+    for sp in a.strain_protocol.filter(enabled=True):
+        strain = sp.strain
+        protocol = sp.protocol
+        for t in protocol.treatments.all():
+            tr = t.treatment
+            ret[str(tr)] = {'name': str(tr)}
+    return ret
+
+
+def concentrations(a):
+    ret = {}
+    for sp in a.strain_protocol.filter(enabled=True):
```

Comment:
strain_protocol, also this feels like there is probably some easy steps to remove this duplication of code